connectors/kibana.py

```diff
@@ -114,14 +116,19 @@ async def prepare(service_type, index_name, config):
         }
 
         logger.info(f"Prepare {CONNECTORS_INDEX}")
-        await es.prepare_index(CONNECTORS_INDEX, docs=[doc], delete_first=True)
+        await es._override_index(CONNECTORS_INDEX, docs=[doc])
```

Comment:
why did you rename it? if it's used outside the class, it should be a public name. why not keep `prepare_index` ? or rename it to `override_index` ?

Reply:
Now it's a different method. They serve pretty different purpose:

1. `prepare_index` is supposed to be called on content indices to check if they have any mappings. If they do not have any mappings, a valid expected mapping is applied to it, nothing else.
2. `_override_index` (name arbitrary, didn't think too much about it) is supposed to be called only by testing code (therefore, the underscore) to simulate Enterprise Search/Kibana creating _any_ index from scratch (connectors index, jobs index, content index). This method also tries to insert some data in its original implementation and I've kept this.

These 2 methods _can_ use same underlying method, of course, but their logic barely intersects and thus I've created 2 different methods.

Reply:
Thinking more about this, method `_override_index` can actually be named `upsert_index` and whole method can be moved to `kibana.py`. 

For `prepare_index` I'd add a validation to check that it's called only on indices that start with `search-*`.

Reply:
ok I think I see what bother me:

since `_override_index` is to be used purely in `kibana.py`, I advise removing it from `ElasticServer` completely.

and then have a `ElasticTestServer` that inherits from `ElasticServer` and implements its own `prepare_index` that you can use in `kibana.py` ibly or elsewhere in tests

Because `_override_index` is to never be run in production, and it's still in the execution path in production even if behind a flag

Reply:
I moved the method fully to Kibana in the end - inheritance hierarchy did not make much sense now + I saw `kibana.py` just using elasticsearch client directly.

What I did not like about it also that this way of implementing `prepare_index` differently in `ElasticTestServer` is:

- Methods have different signature
- It breaks Liskov principle by drastically changing the behaviour of the service with an override.

It can be addressed later so that `kibana.py` wraps those in a class, but for now probably that's good enough.